src/services/pasteEdits.ts

```diff
@@ -0,0 +1,123 @@
+import { addRange } from "../compiler/core";
+import {
+    CancellationToken,
+    SourceFile,
+    Statement,
+    SymbolFlags,
+    TextRange,
+    UserPreferences,
+} from "../compiler/types";
+import { getLineOfLocalPosition } from "../compiler/utilities";
+import {
+    codefix,
+    Debug,
+    fileShouldUseJavaScriptRequire,
+    forEachChild,
+    formatting,
+    getQuotePreference,
+    isIdentifier,
+    textChanges,
+} from "./_namespaces/ts";
+import { addTargetFileImports } from "./refactors/helpers";
+import {
+    addExportsInOldFile,
+    getExistingLocals,
+    getUsageInfo,
+} from "./refactors/moveToFile";
+import {
+    CodeFixContextBase,
+    FileTextChanges,
+    LanguageServiceHost,
+    PasteEdits,
+} from "./types";
+
+const fixId = "providePostPasteEdits";
+/** @internal */
+export function pasteEditsProvider(
+    targetFile: SourceFile,
+    pastedText: string[],
+    pasteLocations: TextRange[],
+    copiedFrom: { file: SourceFile; range: TextRange[]; } | undefined,
+    host: LanguageServiceHost,
+    preferences: UserPreferences,
+    formatContext: formatting.FormatContext,
+    cancellationToken: CancellationToken,
+): PasteEdits {
+    const changes: FileTextChanges[] = textChanges.ChangeTracker.with({ host, formatContext, preferences }, changeTracker => pasteEdits(targetFile, pastedText, pasteLocations, copiedFrom, host, preferences, formatContext, cancellationToken, changeTracker));
+    return { edits: changes, fixId };
+}
+
+function pasteEdits(
+    targetFile: SourceFile,
+    pastedText: string[],
+    pasteLocations: TextRange[],
+    copiedFrom: { file: SourceFile; range: TextRange[]; } | undefined,
+    host: LanguageServiceHost,
+    preferences: UserPreferences,
+    formatContext: formatting.FormatContext,
+    cancellationToken: CancellationToken,
+    changes: textChanges.ChangeTracker,
+) {
+    const statements: Statement[] = [];
+
+    let start = 0;
+    let newText = "";
+    pasteLocations.forEach((location, i) => {
+        if (i === pasteLocations.length - 1) {
+            newText += targetFile.text.slice(start, location.pos) + pastedText[i] + targetFile.text.slice(location.end);
+        }
+        else {
+            newText += targetFile.text.slice(start, location.pos) + pastedText[i];
+            start = location.end;
+        }
+    });

```

Comment:
I think you can avoid this if/else and `start` tracking by applying the edits in reverse order (sorted by insertion position).